cookbook/controller/error_pages.rst

```diff
@@ -5,172 +5,157 @@
+   (``error.html.twig``).
 
-How the Template for the Error and Exception Pages Is Selected
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. _overriding-or-adding-templates:
 
-The TwigBundle contains some default templates for error and
-exception pages in its ``Resources/views/Exception`` directory.
+To override these templates, simply rely on the standard Symfony method for
+:ref:`overriding templates that live inside a bundle <overriding-bundle-templates>`.
+For example, to override the 404 error template for HTML pages, create a new
+``error404.html.twig`` template located at ``app/Resources/TwigBundle/views/Exception/``:
 
-.. tip::
+.. code-block:: html+jinja
 
-    In a standard Symfony installation, the TwigBundle can be found at
-    ``vendor/symfony/symfony/src/Symfony/Bundle/TwigBundle``. In addition
-    to the standard HTML error page, it also provides a default
-    error page for many of the most common response formats, including
-    JSON (``error.json.twig``), XML (``error.xml.twig``) and even
-    JavaScript (``error.js.twig``), to name a few.
+    {# app/Resources/TwigBundle/views/Exception/error404.html.twig #}
+    {% extends 'base.html.twig' %}
 
-Here is how the ``ExceptionController`` will pick one of the
-available templates based on the HTTP status code and request format:
+    {% block body %}
+        <h1>Page not found</h1>
 
-* For *error* pages, it first looks for a template for the given format
-  and status code (like ``error404.json.twig``);
+        <p>
+            The requested page couldn't be located. Checkout for any URL
+            misspelling or <a href="{{ path('homepage') }}">return to the homepage</a>.
+        </p>
+    {% endblock %}
 
-* If that does not exist or apply, it looks for a general template for
-  the given format (like ``error.json.twig`` or
-  ``exception.json.twig``);
+Commonly, Symfony applications redefine the ``error404.html.twig`` template, the
+``error500.html.twig`` template for internal server errors and the generic
+``error.html.twig`` template to catch any other error different from 404 and 500.
 
-* Finally, it ignores the format and falls back to the HTML template
-  (like ``error.html.twig`` or ``exception.html.twig``).
+In case you need them, the ``ExceptionController`` passes some information to
+the error template via the ``status_code`` and ``status_text`` variables that
+store the HTTP status code and message respectively.
 
 .. tip::
 
-    If the exception being handled implements the
+    If your application defines custom exceptions and they implement the
     :class:`Symfony\\Component\\HttpKernel\\Exception\\HttpExceptionInterface`,
-    the ``getStatusCode()`` method will be
-    called to obtain the HTTP status code to use. Otherwise,
-    the status code will be "500".
+    the ``status_code`` variable will contain the value returned by the
+    ``getStatusCode()`` method. Otherwise, the ``status_code`` variable will be ``500``.
```

Comment:
this sentence (if I can read it correctly due to the crazy diffs) is not very great to read. What about:

```rst
You can customize the status code by implementing
:class:`Symfony\\Component\\HttpKernel\\Exception\\HttpExceptionInterface`
and its required ``getStatusCode()`` method. Otherwise, the ``status_code`` will default
to ``500``.
```